video/out/gpu/video.c

```diff
@@ -2473,19 +2475,48 @@ static void pass_scale_main(struct gl_video *p)
 // rendering)
 // If OSD is true, ignore any changes that may have been made to the video
 // by previous passes (i.e. linear scaling)
-static void pass_colormanage(struct gl_video *p, struct mp_colorspace src, bool osd)
+static void pass_colormanage(struct gl_video *p, struct mp_colorspace src,
+                             struct mp_colorspace fbo_colorspace,
+                             bool osd)
 {
     struct ra *ra = p->ra;
 
-    // Figure out the target color space from the options, or auto-guess if
-    // none were set
+    // Configure the destination according to the FBO color space,
+    // unless specific transfer function, primaries or target peak
+    // is set. If values are set to _AUTO, the most likely intended
+    // values are guesstimated later in this function.
     struct mp_colorspace dst = {
-        .gamma = p->opts.target_trc,
-        .primaries = p->opts.target_prim,
+        .gamma = p->opts.target_trc == MP_CSP_TRC_AUTO ?
+                  fbo_colorspace.gamma : p->opts.target_trc,
+        .primaries = p->opts.target_prim == MP_CSP_PRIM_AUTO ?
+                     fbo_colorspace.primaries : p->opts.target_prim,
         .light = MP_CSP_LIGHT_DISPLAY,
-        .sig_peak = p->opts.target_peak / MP_REF_WHITE,
+        .sig_peak = !p->opts.target_peak ?
+                    fbo_colorspace.sig_peak :
+                    p->opts.target_peak / MP_REF_WHITE,
     };
 
+    if (!p->colorspace_override_warned &&
+        ((fbo_colorspace.gamma != MP_CSP_TRC_AUTO &&
+          dst.gamma != fbo_colorspace.gamma) ||
```

Comment:
Could put this onto a single line as `dst.gamma && dst.gamma != fbo_colorspace.gamma` to save a bit on the headache of parsing this conditional.

Reply:
Did exactly that :) .